nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -466,8 +466,7 @@ Costmap2DROS::updateMap()
       layered_costmap_->updateMap(x, y, yaw);
 
       auto footprint = std::make_unique<geometry_msgs::msg::PolygonStamped>();
-      footprint->header.frame_id = global_frame_;
-      footprint->header.stamp = now();
+      footprint->header = pose.header;
```

Comment:
So far, I'm not seeing any actual changes to behavior / code other than right here, where the stamp goes from now() to Time(0)

Reply:
It doesn't go to Time(0). It's filled with actual pose stamp by tf inside `getRobotPose()` method.

Reply:
So `pose` gets its header from getting the robot pose, which is calling https://github.com/ros-planning/navigation2/blob/cf10abebfee60fb5cf09df6836b24594b68c083c/nav2_util/src/robot_utils.cpp#L49, does this actually populate a header stamp? Because you hand it a default constructed pose stamped (https://github.com/ros-planning/navigation2/blob/cf10abebfee60fb5cf09df6836b24594b68c083c/nav2_util/src/robot_utils.cpp#L49) and I'm not sure TF will actually populate that. Please verify that

Reply:
Of course it works like that. I'm not going to formulate any proofs here for verification, please spend some time expanding your horizons yourself with these basic ROS elements, it will be beneficial for you and this project.

Reply:
There is no reason to get nasty. There is a massive amount of code I have to keep track of spanning most robotics domains, its not a reasonable expectation that I remember off the top of my head how every single function is internally implemented at all times, even having read TF2 many times. 

You're adding a reliance on an internal implementation detail of an external library. I'm simply asking for verification.

Reply:
I'm sorry, I got tired of repetitive persuasion about that the update actually does something. If simple 'yes' was enough for you then I could have gone just with it.